### PR TITLE
Auto create security group for internal project traffic

### PR DESCRIPTION
**Pull request by pchico83**

### CODE REVIEW

File: cli/providers/aws/ec2/securitygroups.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	description := fmt.Sprintf("Security Group for environment %s", e.Name)
+	name := fmt.Sprintf("%s-i2kit", e.Name)
+	csgi := &ec2.CreateSecurityGroupInput{
+		Description: &description,
+		GroupName:   &name,
+		VpcId:       &vpc,
+	}
+	sg, err := svc.CreateSecurityGroup(csgi)
+	if err != nil {
+		if !strings.Contains(err.Error(), "InvalidGroup.Duplicate") {
```
Comment (rberrelleza) [inline cli/providers/aws/ec2/securitygroups.go:29]: do the SG name needs to be unique at the VPC level or account level?

File: cli/providers/aws/ec2/securitygroups.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			return err
+		}
+		dsgi := &ec2.DescribeSecurityGroupsInput{
+			GroupNames: []*string{&name},
+		}
+		sgs, err2 := svc.DescribeSecurityGroups(dsgi)
+		if err2 != nil {
+			return err2
+		}
+		for _, i := range sgs.SecurityGroups {
```
Comment (rberrelleza) [inline cli/providers/aws/ec2/securitygroups.go:39]: do we need to handle the case where DescribeSecurityGroups doesn't return the group? (e.g. if there's a group with the same name, but on a different vpc)

File: cli/providers/aws/ec2/securitygroups.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		e.Provider.SecurityGroup = *sg.GroupId
+	}
+
+	asgii := &ec2.AuthorizeSecurityGroupIngressInput{
+		GroupName:               &name,
+		SourceSecurityGroupName: &name,
+	}
+	protocol := "-1"
+	if _, err = svc.AuthorizeSecurityGroupIngress(asgii); err != nil {
+		if strings.Contains(err.Error(), "InvalidParameterValue") {
```
Comment (rberrelleza) [inline cli/providers/aws/ec2/securitygroups.go:58]: what's the escenario that this check is handling?

File: cli/providers/aws/ec2/securitygroups.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	asgii = &ec2.AuthorizeSecurityGroupIngressInput{
+		FromPort:   &ports,
+		ToPort:     &ports,
+		GroupName:  &name,
+		IpProtocol: &protocol,
+		CidrIp:     &cidrIP,
+	}
+	if _, err = svc.AuthorizeSecurityGroupIngress(asgii); err != nil {
+		if strings.Contains(err.Error(), "InvalidParameterValue") {
+			asgii.SetGroupName("")
```
Comment (rberrelleza) [inline cli/providers/aws/ec2/securitygroups.go:90]: what is this name for?

File: cli/providers/aws/ec2/securitygroups.go (hunk 1 above)
Comment (pchico83) [reply cli/providers/aws/ec2/securitygroups.go]: Probably depends on VPC-EC2 or EC2 classic. Would it affect to this logic?

Comment (pchico83) [reply cli/providers/aws/ec2/securitygroups.go:29]: VPC level or EC2-Classic level

File: cli/providers/aws/ec2/securitygroups.go (hunk 2 above)
Comment (pchico83) [reply cli/providers/aws/ec2/securitygroups.go:39]: No, it wouldn't fail on that case

File: cli/providers/aws/ec2/securitygroups.go (hunk 3 above)
Comment (pchico83) [reply cli/providers/aws/ec2/securitygroups.go:58]: Calling a VPC different than EC2-Classic or default VP. I didn't find a better way to detect this scenario.

File: cli/providers/aws/ec2/securitygroups.go (hunk 4 above)
Comment (pchico83) [reply cli/providers/aws/ec2/securitygroups.go:90]: This parameter is not supported in new VPCs (it is only supported by EC2).

**Pull request merged**